### DISCUSSION TITLE
which version of OpenGL axmol uses?

### DISCUSSION THREAD

**Discussion by 10100010010001001010** [axmolengine/axmol#1096] [General]
cocos2d-x uses opengl ES 2.0, is axmol based on it too?

**Comment by aismann** [comment]
> cocos2d-x uses opengl ES 2.0, is axmol based on it too?

Please read the https://github.com/axmolengine/axmol/blob/dev/CHANGELOG

**Reply by 10100010010001001010** [reply]
- **OpenGL** Error 502 in Hole Demo
- Added EGL_RENDERABLE_TYPE to **OpenGL** attributes
- Last test case of **OpenglTest** crashed
- Fixed crash in **OpenGLTest**
- **OpenGL** Error 502 in Hole Demo
- Removes unused files after using glfw3 to create **opengl** context
- Removes unused files for MAC platform after using glfw3 to create **opengl** context.
- DrawNode: fix random crash because of init **opengl** buffer wrongly 
- Label: shadow effect cause **OpenGL** error on iOS
- 3d: Added **opengl** version project/unproject function in camera
- **OpenGL**: Fixed shader-basic and OpenGL testcase 'center' uniform error
- **OpenGL**: Fixed ShaderRetroEffect random crash issue
- **OpenGL**: Fixed offset on retina screen in shaderTest
- **OpenGL**: Fixed shader test crash on android device
- **OpenGL**: Fixed retro effect pos bug in ShaderTest
- **OpenGL**: Fixed a bug of ui::WebView callback operate in OpenGL
- **OpenGL**: Added check for glfwCreateWindow
- iOS: may crashed if multiple **OpenGL ES** contexts coexist

There is only one mention of ES here, everything else is just OpenGL, and nowhere is there any mention of a version of either. I saw somewhere that both OpenGL and OpenGL ES are used, depending on which platform we are making the game for

**Comment by rh101** [comment]
If you search the code, you'll find several references to OpenGL ES 2 on Android and iOS, and OpenGL on Windows, Linux and MacOS

**Comment by aismann** [comment]
I can find this sentence on the cocos2dx 4.0 readme https://github.com/cocos2d/cocos2d-x/blob/v4/README.md

`- OpenGL ES 2.0 (mobile) / OpenGL 2.1 (desktop) / metal(macos and iOS) based`

and here the axmol docu link (there be also):
https://axmolengine.github.io/axmol/
